### PR TITLE
Fix behavior of `Ray2.intersectPoint()`

### PR DESCRIPTION
**Pull request by player-03**
All points to the right of the ray produce a negative dot product. Without `Math.abs()`, these points are considered to be intersecting, because negative numbers are always less than `tolerance`.

### CODE REVIEW

File: src/hxmath/geom/Ray2.hx [hunk 1]
```diff
@@ -89,4 +89,4 @@ class Ray2
         
         return hitIndex;
     }
-}
```
Comment (tbrosman) [inline src/hxmath/geom/Ray2.hx:92]: Minor: can you revert this block?

**Comment by tbrosman** [comment]
Nice find! That was definitely wrong.

File: src/hxmath/geom/Ray2.hx (hunk 1 above)
Comment (player-03) [reply src/hxmath/geom/Ray2.hx]: It seems trailing newlines are so standard nowadays that the GitHub editor automatically adds them.

**Comment by tbrosman** [comment]
Also: can you add a test for this case? Should be pretty simple, but a good sanity-check.

**Comment by player-03** [comment]
[It's in my other pull request](https://github.com/tbrosman/hxmath/pull/79/files#diff-25cc0838f97dce7bd7e4c2856ad5274c1ed4df377048ac07d890bc6d657fcc14R47). I'll uncomment it once this gets merged.

**Comment by tbrosman** [comment]
Done. Thanks!

**Pull request merged**